publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -269,6 +271,42 @@ public Resource getResourceByIdentifier(SortableIdentifier identifier) throws No
         return readResourceService.getResourceByIdentifier(identifier);
     }
 
+    public Optional<Resource> getResourceAndFilesByIdentifier(SortableIdentifier identifier) {
+        var partitionKey = resourceQueryObject(identifier).toDao().getByTypeAndIdentifierPartitionKey();
+        var queryRequest = new QueryRequest()
+                               .withTableName(tableName)
+                               .withIndexName(BY_TYPE_AND_IDENTIFIER_INDEX_NAME)
+                               .withKeyConditionExpression("#PK3 = :value")
+                               .withExpressionAttributeNames(Map.of("#PK3", "PK3"))
+                               .withExpressionAttributeValues(Map.of(":value", new AttributeValue(partitionKey)));
+
+        var entries = client.query(queryRequest).getItems().stream()
+                          .map(map -> parseAttributeValuesMap(map, Dao.class))
+                          .toList();
+
+        var resource = extractResource(entries);
+        var files = extractFiles(entries);
+
+        return resource.map(res -> res.copy().withFiles(files).build());
+    }
```

Comment:
Fetching Resource with files by single call. Returning Optional instead of throwing ApiGateway NotFoundException, so we handle specific behavior in handlers/services.